engines/dfc_provider/app/services/dfc_provider/authorization_control.rb

```diff
@@ -8,24 +8,55 @@ def initialize(access_token)
       @access_token = access_token
     end
 
+    def safe_process
+      process
+    rescue JWT::VerificationError, JWT::ExpiredSignature, JWT::ImmatureSignature => e
+      # To help development debugging
+      raise e if Rails.env.development?
+
+      nil
+    end
+
     def process
       decode_token
+      control_payload
       find_ofn_user
     end
 
+    private
+
     def decode_token
       data = JWT.decode(
         @access_token,
         nil,
-        false
+        true,
+        { algorithms: ['RS256'], jwks: jwks_hash }
       )
 
       @header = data.last
       @payload = data.first
     end
 
     def find_ofn_user
-      Spree::User.where(email: @payload['email']).first
+      Spree::User.where(provider: 'openid_connect', uid: @payload['email']).first
+    end
+
+    def control_payload
+      raise 'Email Not Found' if @payload['email'].blank?
+    end
+
+    def client_secret
+      Devise.omniauth_configs[:openid_connect].options[:client_options][:secret]
+    end
```

Comment:
This is dead code, right? It's a private method which isn't called from anywhere. Should it be called from somewhere or can it be deleted?

Reply:
Ah right, good catch,this method is not use anymore, cleaning it.